### PR TITLE
[New Extension] Summarise youtube videos with AI 

### PR DESCRIPTION
**Pull request by iKasch**
## Description

By using this extension, you can obtain a brief summary of any YouTube video. The extension operates by extracting the transcript in the background and creating bullet-point summaries. In case the transcript is lengthy, the summarization process is divided into multiple segments.

You have to choose bewteen Raycast AI or ChatGPT. If you choose ChatGPT an OpenAI API Key is needed.

**Command:**

| command | description |
|----------|----------|
| `summarizeVideo` | provide a youtube link or ID to get the summary as a bullet point list. It will also get some basic video infos like `title`, `channelName`, `publishDate`, `viewCount`, `thumbnail` and `length`. |

## Screencast

CleanShot 2023-05-15 at 02 01 59@2x

summarize-youtube-video-1
summarize-youtube-video-3
summarize-youtube-video-2

https://user-images.githubusercontent.com/2265339/236178043-889c6dad-7449-4013-9aa6-64d7d3802085.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/summarize-youtube-video-with-chatgpt/src/getAiSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          (acc, curr) => {
+            if (acc[acc.length - 1].length + curr.length > SUMMARY_MAX_CHARS) {
+              acc.push(curr);
+            } else {
+              acc[acc.length - 1] += curr;
+            }
+            return acc;
+          },
+          [""]
+        );
```
Comment (JonathanWbn) [inline extensions/summarize-youtube-video-with-chatgpt/src/getAiSummary.tsx:48]: I found an issue with this for a YT where the transcript seemingly had very few `.`s. This resulted in blocks that were too long and produced an error. I added this condition which seems to fix the issue:
```ts
if (curr.length > SUMMARY_MAX_CHARS) {
  const splitTranscription = curr.match(/.{1,12000}/g);
  splitTranscription?.forEach((split) => {
    acc.push(split);
  });
} else if (acc[acc.length - 1]...
```

**Review by JonathanWbn — APPROVED** [review]
I think it's a really cool idea. 👍 

I found one issue and added a suggestion on how it could be addressed 😊

File: extensions/summarize-youtube-video-with-chatgpt/src/getAiSummary.tsx (hunk 1 above)
Comment (iKasch) [reply extensions/summarize-youtube-video-with-chatgpt/src/getAiSummary.tsx:48]: Nice, thank you!

**Comment by iKasch** [comment]
Hey @pernielsentikaer - I see you are reviewing most of the extensions here. Is there anything I can help out with or improve here?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**